galaxy_ng/app/api/v3/serializers/sync.py

```diff
@@ -127,6 +145,60 @@ class Meta:
             'ca_cert': {'write_only': True},
         }
 
+    def to_representation(self, instance):
+        """
+        Splits proxy_url field from DB in 3 fields for representation.
+        proxy_url, proxy_username, proxy_password (write-only)
+        """
+
+        if instance.proxy_url is not None:
+            url, username, _ = strip_auth_from_url(instance.proxy_url)
+            instance.proxy_url = url
+            instance.proxy_username = username
+            # instance.proxy_password is write_only, no need to represent.
+
+        return super().to_representation(instance)
+
+    def save(self):
+        """
+        Rejoins 3 fields url, username, password back to proxy_url on DB.
+
+        If proxy_url is null or blank, the field is cleaned up in DB
+        regardless of username, password values.
+
+        If proxy, url, password or username is missing from the payload then
+        should be the same as the existing value in the DB or None.
+
+        If proxy username/password is set to null or blank, the fields are
+        cleaned up in DB.
+        """
+
+        empty = object()
```

Comment:
Why do you send an empty object instead of just checking if the URL is none in `strip_auth_from_url`?